php/example_code/kms/KmsBasics.php

```diff
@@ -0,0 +1,308 @@
+        if($granteeARN){
+            $operations = [
+                "ENCRYPT",
+                "DECRYPT",
+                "DESCRIBE_KEY",
+            ];
+            $grant = $this->kmsService->createGrant($key['KeyId'], $granteeARN, $operations);
+            echo "The grant Id is: {$grant['GrantId']}\n";
+        }else{
+            echo "Steps 7, 8, and 9 will be skipped.\n";
+        }
+        $this->pressEnter();
+
+        // 8. List grants for the KMS key.
+        if($granteeARN){
+            echo "8. List grants for the KMS key.\n\n";
+            $grantsPaginator = $this->kmsService->listGrants($key['KeyId']);
+            foreach($grantsPaginator as $page){
+                foreach($page['Grants'] as $grant){
+                    echo $grant['GrantId'] . "\n";
+                }
+            }
+        }else{
+            echo "Skipping step 8...\n";
+        }
+        $this->pressEnter();
+
+        // 9. Revoke the grant.
+        if($granteeARN) {
+            echo "\n";
+            echo "9. Revoke the grant.\n";
+            $this->pressEnter();
+            $this->kmsService->revokeGrant($grant['GrantId'], $keyInfo['KeyId']);
+            echo "{$grant['GrantId']} was successfully revoked!\n";
+        }else{
+            echo "Skipping step 9...\n";
+        }
+        $this->pressEnter();
+
+        // 10. Decrypt the data.
+        echo "\n";
+        echo "10. Decrypt the data.\n";
+        echo "Lets decrypt the data that was encrypted before.\n";
+        echo "We'll use the same key to decrypt the string that we encrypted earlier in the program.\n";
+        $this->pressEnter();
+        $decryption = $this->kmsService->decrypt($keyInfo['KeyId'], $encryption['CiphertextBlob'], $encryption['EncryptionAlgorithm']);
+        echo "The decrypted text is: {$decryption['Plaintext']}\n";
+        $this->pressEnter();
+
+        // 11. Replace a Key Policy.
+        echo "\n";
+        echo "11. Replace a Key Policy.\n";
+        echo "A key policy is a resource policy for an KMS key. Key policies are the primary way to control access to KMS keys.\n";
+        echo "Every KMS key must have exactly one key policy. The statements in the key policy determine who has permission to use the KMS key and how they can use it.\n";
+        echo " You can also use IAM policies and grants to control access to the KMS key, but every KMS key must have a key policy.\n";
+        echo "We will replace the key's policy with a new one:\n";
+        $stsClient = new StsClient([]);
+        $result = $stsClient->getCallerIdentity();
+        $accountId = $result['Account'];
+        $keyPolicy = <<< KEYPOLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [{
+        "Effect": "Allow",
+        "Principal": {"AWS": "arn:aws:iam::$accountId:root"},
+        "Action": "kms:*",
+        "Resource": "*"
+    }]
```

Comment:
This strikes me as a weird policy to show, because doesn't the root user already have this permission (I could be wrong about this)? What about narrowing it to just the current user instead?

Reply:
I agree, but since this is based on the Java and that's how it was implemented there, I'll bring it up with Scott and see if there's a better way to do this.